src/components/ListItem.jsx

```diff
@@ -1,5 +1,45 @@
+import { useEffect, useState } from 'react';
+import { updateItem } from '../api/firebase';
+import { ONE_DAY_IN_MILLISECONDS } from '../utils';
 import './ListItem.css';
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+export function ListItem({ listPath, item, name }) {
+	const [isChecked, setIsChecked] = useState(false);
+
+	function handleChange(e) {
+		// ensures checked box can be toggled
+		setIsChecked(!isChecked);
+		// pass listPath, copies the item object and sets a new value named isChecked
+		updateItem(listPath, { ...item, isChecked: !isChecked });
+	}
+
+	// this useEffect will run everytime the value of "dateLastPurchased" on the item object changes
+	useEffect(() => {
+		//
+		if (item.dateLastPurchased) {
+			const now = new Date();
+			// calculate expiration time by adding on 24 hours onto the value that is stored on the dateLastPurchased key
+			// this variable will be in milliseconds
+			const expirationTime =
+				new Date(
+					item.dateLastPurchased.seconds * 1000 +
+						Math.floor(item.dateLastPurchased.nanoseconds / 1000000),
+				).getTime() + ONE_DAY_IN_MILLISECONDS;
+
+			// Set checkbox state will become unchecked when the current date matches the expiration time
+			setIsChecked(now.getTime() < expirationTime);
+		}
+	}, [item.dateLastPurchased]);
+
+	return (
+		<li className="ListItem">
+			<input
+				checked={isChecked}
+				onChange={handleChange}
+				type="checkbox"
+				id="item-checkbox"
```

Comment:
Every list item have the same id `item-checkbox`. This results in the accessibility issue, where clicking on any item label changes the checkbox of the first item. Instead, each label should change the corresponding checkbox. To achieve this, we can simply change the id of the html element to `item.id` in both places:
```
<input ... id={item.id} />
<label ... id={item.id} />
```

Please see the video recording of the bug:

https://github.com/the-collab-lab/tcl-73-smart-shopping-list/assets/88466266/7b6c6351-7fbe-4294-b484-f7d4c8fb2fbf

Reply:
Thanks for spotting this and providing the video. We will use this dynamic approach to make sure the item checked matches up with its label.